velox/expression/CastExpr.cpp

```diff
@@ -980,6 +1029,15 @@ ExprPtr CastCallToSpecialForm::constructSpecialForm(
     std::vector<ExprPtr>&& compiledChildren,
     bool trackCpuUsage,
     const core::QueryConfig& config) {
+  const auto inputKind = compiledChildren[0]->type()->kind();
```

Comment:
Shouldn't this happen after checking that compiledChildren.size() == 1 on L1041? 

Might be safer to do that in CastExpr's constructor.

Reply:
Thanks for the catch.
> Might be safer to do that in CastExpr's constructor.

SparkCastExpr extends CastExpr, so the check would impact Spark as well if it's in the constructor of CastExpr.